packages/nameguard-sdk/src/index.ts

```diff
@@ -462,16 +462,30 @@ export interface SecurePrimaryNameResult {
   nameguard_result: NameGuardReport | null;
 }
 
-// TODO: Let's apply more formalization to this error class.
-class NameGuardError extends Error {
+export enum NameGuardErrorType {
+  Server = "Server",
+  Network = "Network",
+  Timeout = "Timeout",
+  Abort = "Abort",
+  Input = "Input",
+  Unknown = "Unknown",
+}
+
+export class NameGuardError extends Error {
   constructor(
-    public status: number,
+    public type: NameGuardErrorType,
+    public status?: number,
```

Comment:
What's the goal for this `status` field?

Reply:
he HTTP status code associated with the error, if applicable. I've updated it to include a doc block.